### PR TITLE
Added rabbitmq:dynamic-consumer command

### PR DESCRIPTION
**Pull request by tiborb**
A context-aware command: https://github.com/videlalvaro/RabbitMqBundle/issues/264

### CODE REVIEW

**Comment by tiborb** [comment]
I can also imagine making the --context argument optional. 
In that case the command's usage would be like a "normal" consumer with the difference that the queue options are loaded only from a service which takes a context argument.

File: Command/BaseConsumerCommand.php [hunk 1]
```diff
@@ -87,14 +87,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if (0 > $this->amount) {
             throw new \InvalidArgumentException("The -m option should be null or greater than 0");
         }
-
+        $this->initConsumer($input);
```
Comment (videlalvaro) [inline Command/BaseConsumerCommand.php]: why do you need these changes here?

Comment (tiborb) [inline Command/BaseConsumerCommand.php]: I have separated the consumer initialisation from the actual execution.

It is needed for extending the BaseConsumerCommand by the DynamicConsumerCommand and allows me to pass an argument to the dynamic consumer (the context in this case). 
It also makes the code more testable and easier to reuse.

Comment (videlalvaro) [inline Command/BaseConsumerCommand.php]: How does this interplay with other features. Does it introduce any BC breaks?

Comment (tiborb) [inline Command/BaseConsumerCommand.php]: No BC here.
Only the execute calls the init method, so there is no change in the standard consumer's  workflow.

**Comment by videlalvaro** [comment]
thanks!

**Pull request merged**